.github/workflows/ibis-backends.yml

```
@@ -318,7 +318,7 @@ jobs:
       - uses: syphar/restore-pip-download-cache@v1
         with:
           requirement_files: poetry.lock
-          custom_cache_key_element: ${{ steps.install_python.outputs.python-version }}
+          custom_cache_key_element: ${{ matrix.os }}-${{ steps.install_python.outputs.python-version }}
```

```suggestion
          custom_cache_key_element: ${{ steps.install_python.outputs.python-version }}
```
Is this necessary? Cache can't conflict across different OS executors, I think, [especially because they're cached to different directory base don OS](https://github.com/syphar/restore-pip-download-cache/blob/main/src/utils.ts#L22-L36).

---

> It means either the cached content was corrupted or there is an edge scenario this can break (e.g., what if you change from a specific os version to latest)

If I had to guess, it's more likely the latter, so just changing the cache key circumvented it. In which case, I think just changing it helped. I think it's worth trying to revert this, but happy to let somebody else comment first.